### PR TITLE
Feature/#47 validate nothing forgotten

### PR DESCRIPTION
**Pull request by GitRon**
Hi @lociii and @costela 

I implemented the three features.

1. Management command to check for missing fields (to be used locally or in the CI pipeline)
2. Strict mode which prohibits scrubbing if fields are not set
3. scrubbers.Keep to be able to mark a field as being scrubbed but still keep its value

I added two new settings variables in addition to the strict mode for cuostmisation.

What do you think so far?

Best from Cologne  
Ronny

### CODE REVIEW

File: django_scrubber/__init__.py [hunk 1]
```diff
@@ -11,6 +12,16 @@
     'SCRUBBER_ADDITIONAL_FAKER_PROVIDERS': [],
     'SCRUBBER_FAKER_LOCALE': None,
     'SCRUBBER_MAPPING': dict(),
+    'SCRUBBER_STRICT_MODE': False,
+    'SCRUBBER_REQUIRED_FIELD_TYPES': (models.CharField, models.TextField, models.URLField, models.JSONField),
```
[the remark this answers is not in the document]
Comment (GitRon) [reply django_scrubber/__init__.py:16]: Hi @lociii 

agree on Email and IPField - the birthday should only bei PII if its connected to a name. But why not. File and image fields will only contain links - they are not relevant. Though, they will be quite useless if you just dump the database.

What do you think?
Ronny

[the remark this answers is not in the document]
Comment (lociii) [reply django_scrubber/__init__.py:16]: Then let's add email and IP to not annoy people too much and still be safe enough.

[the remark this answers is not in the document]
Comment (GitRon) [reply django_scrubber/__init__.py:16]: 👌

[the remark this answers is not in the document]
Comment (GitRon) [reply django_scrubber/__init__.py]: done.

File: README.md [hunk 2]
```diff
@@ -230,6 +289,28 @@ Set an alternative locale for Faker used during the scrubbing process.
...
+`SCRUBBER_REQUIRED_FIELD_TYPES`. If you have unscrubbed fields and this flag is active, you can't run
+`pyhton manage.py scrub_data`.
+
+### `SCRUBBER_REQUIRED_FIELD_TYPES`:
+
+Defaults to all text-based Django model fields. Usually, privacy-relevant data is only stored in text-fields, numbers
+and booleans (usually) can't contain sensitive personal data. These fields will be checked when running
+`python manage.py scub_validation`.
+
+(default: (models.CharField, models.TextField, models.URLField, models.JSONField))
```
Comment (lociii) [inline README.md:304]: Please adjust to the updated list or just reference where to find the default set.

File: django_scrubber/__init__.py [hunk 3]
```diff
@@ -11,6 +12,17 @@
     'SCRUBBER_ADDITIONAL_FAKER_PROVIDERS': [],
     'SCRUBBER_FAKER_LOCALE': None,
     'SCRUBBER_MAPPING': dict(),
+    'SCRUBBER_STRICT_MODE': False,
+    'SCRUBBER_REQUIRED_FIELD_TYPES': (models.CharField, models.TextField, models.URLField, models.JSONField,
+                                      models.GenericIPAddressField, models.EmailField),
```
Comment (lociii) [inline django_scrubber/__init__.py:17]: Can you please add a trailing comma? This should reformat and move all items in their own line.

**Review by lociii — COMMENTED** [review]
Last polishing.

File: README.md (hunk 2 above)
Comment (GitRon) [reply README.md]: 👌

**Comment by GitRon** [comment]
Done 👌

**Pull request merged**